Patch taken from
  https://forums.developer.nvidia.com/t/455-23-04-page-allocation-failure-in-kernel-module-at-random-points/155250/55
Bug: https://bugs.gentoo.org/755497
Credits go to Gregory Beauregard for pointing this out

--- a/kernel/nvidia-modeset/nvidia-modeset-linux.c
+++ b/kernel/nvidia-modeset/nvidia-modeset-linux.c
@@ -282,8 +282,8 @@
  * are called while nvkms_lock is held.
  *************************************************************************/
 
-/* Don't use kmalloc for allocations larger than 128k */
-#define KMALLOC_LIMIT (128 * 1024)
+/* Don't use kmalloc for allocations larger than one page */
+#define KMALLOC_LIMIT PAGE_SIZE
 
 void* NVKMS_API_CALL nvkms_alloc(size_t size, NvBool zero)
 {
